### PR TITLE
Remove unneeded variable from addresses.py

### PR DESCRIPTION
**Pull request by cis-muzahid**
Remove unneeded variable from addresses.py

### CODE REVIEW

File: src/tests/test_addresses.py [hunk 1]
```diff
@@ -10,6 +10,16 @@
 class TestAddresses(unittest.TestCase):
     """Test addresses manipulations"""
 
+    def test_encodeBase58(self):
+        """ Encode base alphabets and cover with 0 number """
+        self.assertEqual(
+            addresses.encodeBase58(0), "1", "Error in asserting")
+
```
Comment (PeterSurda) [inline src/tests/test_addresses.py:17]: encode 1: 2

File: src/tests/test_addresses.py [hunk 2]
```diff
@@ -10,6 +10,16 @@
...
+    def test_decodeBase58(self):
+        """ Decode a Base X encoded string """
+        self.assertEqual(
+            addresses.decodeBase58("0"), 0, "Error in asserting")
+
```
Comment (PeterSurda) [inline src/tests/test_addresses.py:22]: decode "1": 0

File: src/tests/test_addresses.py [hunk 3]
```diff
@@ -33,3 +49,25 @@ def test_encode(self):
...
+        self.assertEqual(addresses.encodeVarint(254), change_integer)
+
+        change_integer = pack('>B', 254) + pack('>I', 65536)
+
+        self.assertEqual(addresses.encodeVarint(65536), change_integer)
+
+        change_integer = pack('>B', 255) + pack('>Q', 4294967296)
+
+        self.assertEqual(addresses.encodeVarint(4294967296), change_integer)
```
Comment (g1itch) [inline src/tests/test_addresses.py:72]: I covered those here: https://github.com/Bitmessage/PyBitmessage/pull/1788/files#diff-74690536f227be20ffb8cae3d57e83ea4b40945516c3798cb26d018e61b20fa2

File: src/tests/test_addresses.py [hunk 4]
```diff
@@ -5,11 +5,27 @@
...
+
+        self.assertEqual(
+            addresses.encodeBase58(1), "2", "Error in asserting")
+
+    def test_decodeBase58(self):
+        """ Decode a Base X encoded string """
+        self.assertEqual(
+            addresses.decodeBase58("0"), 0, "Error in asserting")
+        self.assertEqual(
+            addresses.decodeBase58("1"), 0, "Error in asserting")
```
Comment (g1itch) [inline src/tests/test_addresses.py:27]: I covered it here: https://github.com/Bitmessage/PyBitmessage/pull/1788/files#diff-41b1875b15c45657de6a21377434a4352f6da8d40330451e4824dd16e3646905

File: src/tests/test_addresses.py [hunk 5]
```diff
@@ -5,11 +5,27 @@
...
+    def test_encodeBase58(self):
+        """ Encode base alphabets and cover with 0 number """
+        self.assertEqual(
+            addresses.encodeBase58(0), "1", "Error in asserting")
+
+        self.assertEqual(
+            addresses.encodeBase58(1), "2", "Error in asserting")
+
+    def test_decodeBase58(self):
+        """ Decode a Base X encoded string """
```
Comment (g1itch) [inline src/tests/test_addresses.py:23]: No Base X after your changes in `addresses`. Now it's only base58.

File: src/tests/test_addresses.py [hunk 6]
```diff
@@ -5,11 +5,27 @@
...
 
 from .samples import sample_address, sample_ripe
+from struct import pack
 
 
 class TestAddresses(unittest.TestCase):
     """Test addresses manipulations"""
 
+    def test_encodeBase58(self):
+        """ Encode base alphabets and cover with 0 number """
```
Comment (g1itch) [inline src/tests/test_addresses.py:15]: This is a weird string with redundant spaces.

File: src/tests/test_addresses.py [hunk 7]
```diff
@@ -5,11 +5,27 @@
 from pybitmessage import addresses
 
 from .samples import sample_address, sample_ripe
+from struct import pack
```
Comment (PeterSurda) [inline src/tests/test_addresses.py:8]: remove this file

**Review by PeterSurda — CHANGES_REQUESTED** [review]
remove `test_addresses.py`

File: src/addresses.py [hunk 8]
```diff
@@ -1,281 +1,34 @@
...
-        elif len(embeddedRipeData) < 4:
-            return 'ripetooshort', 0, 0, ''
-        x00string = b'\x00' * (20 - len(embeddedRipeData))
-        return status, addressVersionNumber, streamNumber, \
-            x00string + embeddedRipeData
-
-
-def addBMIfNotPresent(address):
-    """Prepend BM- to an address if it doesn't already have it"""
-    address = str(address).strip()
```
Comment (g1itch) [inline src/addresses.py:280]: LOL, you've removed an addresses manipulation part of the protocol implementation (;

**Comment by g1itch** [comment]
This is now about removing the kwarg. So you may just rebase those redundant commits:
```
$ git checkout test_address
$ git fetch upstream
$ git merge-base base58-nokwarg upstream/v0.6
1bf314545541f10a3284634f1dae3925cfc44262
$ git rebase -i 1bf314545541f10a3284634f1dae3925cfc44262

d 30286d1f ignore coverage files
pick 53196b85 cover two methods in coverage
s 876d9dbb remove passing alphabes using same file and fix issue
s d6070aae add more coverage
s 5ecafef6 add more coverage
s 034936e5 fixing code quality issue
s 03bd50ea revert changes
s f5b820b5 remove new test cases and restore address
s b94986e0 changes made older
s a71d041a remove import
s 627cf4ec changes done

$ git restore --staged .gitignore
$ git restore .gitignore
$ git rebase --continue

>> remove passing alphabet into encodeBase58 and decodeBase58 in addresses (do ^k ^k ^k ^k for the rest messages)
```

The [result](https://github.com/g1itch/PyBitmessage/commits/base58-nokwarg)

**Comment by PeterSurda** [comment]
@g1itch how about I add another check to limit the number of commits per PR to 3?

**Comment by g1itch** [comment]
> @g1itch how about I add another check to limit the number of commits per PR to 3?

As for me it's a very strange decision. The argumentation is the same as I wrote in #1828: any change more complex than removing a couple of spaces may require more than 3 commits. #1389 or #1794 are not fit to such limit. When you're solving different aspects of the same problem it's better to do it in the separate commits in my opinion.

Currently the only wish for the project, I have, is the coherent and orderly development process: when each developer looks into the PRs of others, writes meaningful PR descriptions and commit messages, squashes the commits reverting each other. Thank god, at least everybody started to rebase their branches to v0.6 tip. I receive a dozen of notifications a day and try to look into each PR, but most of them are hard to read. The commit history is also barely readable. That's why I'm suggesting the squashing and rewording.

File: .gitignore [hunk 9]
```diff
@@ -18,3 +18,5 @@ docs/_*/*
 docs/autodoc/
 build
 pyan/
+**.coverage
+**htmlcov*
```
Comment (g1itch) [inline .gitignore:22]: The changes to `.gitignore` definitely should go as a separate PR with motivation. It's clearly unrelated to the base58 code.

**Comment by PeterSurda** [comment]
> As for me it's a very strange decision. The argumentation is the same as I wrote in #1828: any change more complex than removing a couple of spaces may require more than 3 commits. #1389 or #1794 are not fit to such limit. When you're solving different aspects of the same problem it's better to do it in the separate commits in my opinion.

I want the developer to get meaningful automated feedback. If it can't be automated, then I need to do it manually, and that's error prone and time consuming.

> Currently the only wish for the project, I have, is the coherent and orderly development process: when each developer looks into the PRs of others, writes meaningful PR descriptions and commit messages, squashes the commits reverting each other.

We're having a daily meeting where we discuss PRs, the problem is I can't think of everything everytime, and it gets more error prone when I have to review the same PR twenty times. 

> Thank god, at least everybody started to rebase their branches to v0.6 tip.

I wrote a script for that, see https://git.bitmessage.org/PeterSurda/termux_devenv/raw/branch/master/files/git-tools.sh
It checks all your open PRs and if it needs rebasing, rebases them. Set your repo to `origin` and upstream to `upstream` to get it working. Unfortunately another problem appeared, if it happens too frequently, github will hit the API rate limit and they refused to raise it for me even thought I'm on a paid account.

> I receive a dozen of notifications a day and try to look into each PR, but most of them are hard to read. The commit history is also barely readable.

This is a fair point, the problem is this burdens me until the developers are trained properly.

> That's why I'm suggesting the squashing and rewording.

That's a good suggestion but as I said I'd like to automate this check somehow.

**Review by PeterSurda — APPROVED** [review]

**Pull request merged**